scalding-beam/src/main/scala/com/twitter/scalding/beam_backend/KryoCoder.scala

```diff
@@ -0,0 +1,42 @@
+package com.twitter.scalding.beam_backend
+
+import com.twitter.chill.Kryo
+import com.twitter.chill.config.ScalaMapConfig
+import com.twitter.scalding.serialization
+import java.io.{ByteArrayOutputStream, InputStream, OutputStream}
+import org.apache.beam.sdk.coders.{AtomicCoder, CoderException => BCoderException}
+
+final class KryoCoder[T] extends AtomicCoder[T] {
+  override def encode(value: T, os: OutputStream): Unit = {
+    val kryo = KryoCoder.KryoState.get()
+    if (value == null) {
+      throw new BCoderException("cannot encode a null value")
+    }
+    val bos = new ByteArrayOutputStream()
+    val bosOuotput = new com.esotericsoftware.kryo.io.Output(bos)
+    kryo.writeClassAndObject(bosOuotput, value)
+    bosOuotput.flush()
+    os.write(bos.toByteArray.length)
+    os.write(bos.toByteArray)
+    os.flush()
+  }
+  override def decode(is: InputStream): T = {
+    val kryo = KryoCoder.KryoState.get()
+    val size = is.read()
+    val input = new com.esotericsoftware.kryo.io.Input(is, size)
+    kryo.readClassAndObject(input).asInstanceOf[T]
+  }
+}
+
+object KryoCoder{
+  val KryoState: ThreadLocal[Kryo] = {
```

Comment:
I am not a fan of this.

1. note that when you are building the `BeamOp` you have access to the `Config`.
2. we have traditionally configured kryo by using the `Config` to get a `KryoHadoop` instance.

What I would prefer is in the BeamPlanner you build the `KryoInstantiator` and then pass that as needed.

Note: the KyroInsantiator is the abstraction recommended for what you are doing here (building new instances) and KryoPool can safely share Kryo instances and buffers: https://github.com/twitter/chill/blob/develop/chill-java/src/main/java/com/twitter/chill/KryoPool.java

Having kryo be unconfigurable isn't great since not that uncommonly we needed to configure it (e.g. set up thrift/proto/avro support).